frame/referenda/src/migration.rs

```diff
@@ -0,0 +1,230 @@
+		Rejected(Moment, Deposit<AccountId, Balance>, Option<Deposit<AccountId, Balance>>),
+		/// Referendum finished with cancellation. Submission deposit is held.
+		Cancelled(Moment, Deposit<AccountId, Balance>, Option<Deposit<AccountId, Balance>>),
+		/// Referendum finished and was never decided. Submission deposit is held.
+		TimedOut(Moment, Deposit<AccountId, Balance>, Option<Deposit<AccountId, Balance>>),
+		/// Referendum finished with a kill.
+		Killed(Moment),
+	}
+
+	#[storage_alias]
+	pub type ReferendumInfoFor<T: Config<I>, I: 'static> =
+		StorageMap<Pallet<T, I>, Blake2_128Concat, ReferendumIndex, ReferendumInfoOf<T, I>>;
+}
+
+pub mod v1 {
+	use super::*;
+
+	/// The log target.
+	const TARGET: &'static str = "runtime::democracy::migration::v1";
+
+	/// Transforms a submission deposit of ReferendumInfo(Approved|Rejected|Cancelled|TimedOut) to
+	/// optional value, making it refundable.
+	pub struct MigrateV0ToV1<T, I = ()>(PhantomData<(T, I)>);
+	impl<T: Config<I>, I: 'static> OnRuntimeUpgrade for MigrateV0ToV1<T, I> {
+		#[cfg(feature = "try-runtime")]
+		fn pre_upgrade() -> Result<Vec<u8>, &'static str> {
+			let onchain_version = Pallet::<T, I>::on_chain_storage_version();
+			assert_eq!(onchain_version, 0, "migration from version 0 to 1.");
+			let referendum_count = v0::ReferendumInfoFor::<T, I>::iter().count();
+			log::info!(
+				target: TARGET,
+				"pre-upgrade state contains '{}' referendums.",
+				referendum_count
+			);
+			Ok((referendum_count as u32).encode())
+		}
+
+		fn on_runtime_upgrade() -> Weight {
+			let current_version = Pallet::<T, I>::current_storage_version();
+			let onchain_version = Pallet::<T, I>::on_chain_storage_version();
+			let mut weight = T::DbWeight::get().reads(1);
+			log::info!(
+				target: TARGET,
+				"running migration with current storage version {:?} / onchain {:?}.",
+				current_version,
+				onchain_version
+			);
+			if onchain_version != 0 {
+				log::warn!(target: TARGET, "skipping migration from v0 to v1.");
+				return weight
+			}
+			v0::ReferendumInfoFor::<T, I>::iter().for_each(|(key, value)| {
+				let maybe_new_value = match value {
+					v0::ReferendumInfo::Ongoing(_) | v0::ReferendumInfo::Killed(_) => None,
+					v0::ReferendumInfo::Approved(e, s, d) =>
+						Some(ReferendumInfo::Approved(e, Some(s), d)),
+					v0::ReferendumInfo::Rejected(e, s, d) =>
+						Some(ReferendumInfo::Rejected(e, Some(s), d)),
+					v0::ReferendumInfo::Cancelled(e, s, d) =>
+						Some(ReferendumInfo::Cancelled(e, Some(s), d)),
+					v0::ReferendumInfo::TimedOut(e, s, d) =>
+						Some(ReferendumInfo::TimedOut(e, Some(s), d)),
+				};
+				if let Some(new_value) = maybe_new_value {
+					weight.saturating_accrue(T::DbWeight::get().reads_writes(1, 1));
+					log::info!(target: TARGET, "migrating referendum #{:?}", &key);
+					ReferendumInfoFor::<T, I>::insert(key, new_value);
+				} else {
+					weight.saturating_accrue(T::DbWeight::get().reads(1));
+				}
+			});
+			weight
```

Comment:
```suggestion
			});
			
			StorageVersion::new(1).put::<Pallet<T>>();
			
			weight
```

We need to store the version we have migrated to in the storage, this isn't happening automatically.